std/array.d

```diff
@@ -192,13 +192,21 @@ if (isIterable!Range && !isNarrowString!Range && !isInfinite!Range)
 
 /**
 Convert a narrow string to an array type that fully supports random access.
-This is handled as a special case and always returns a $(D dchar[]),
-$(D const(dchar)[]), or $(D immutable(dchar)[]) depending on the constness of
-the input.
+This is handled as a special case and always returns an array of `dchar`
+
+Params:
+    str = `isNarrowString` to be converted to an array of `dchar`
+Returns:
+    a $(D dchar[]), $(D const(dchar)[]), or $(D immutable(dchar)[]) depending on the constness of
+    the input.
 */
-ElementType!String[] array(String)(String str) if (isNarrowString!String)
+@trusted ElementType!String[] array(String)(scope String str) if (isNarrowString!String)
```

Comment:
Applying `@trusted` to a template is really dangerous. What if `toUTF32` becomes unsafe in the future?